go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -220,41 +230,44 @@ func addDefaultConntrackRules(nodeName, gwBridge, gwIntf string) error {
 	// table 0, packets coming from external. Send it through conntrack and
 	// resubmit to table 1 to know the state of the connection.
 	_, stderr, err = util.RunOVSOfctl("add-flow", gwBridge,
-		fmt.Sprintf("priority=50, in_port=%s, ip, "+
-			"actions=ct(zone=%d, table=1)", ofportPhys, config.Default.ConntrackZone))
+		fmt.Sprintf("cookie=%s, priority=50, in_port=%s, ip, "+
+			"actions=ct(zone=%d, table=1)", util.DefaultOpenFlowCookie, ofportPhys, config.Default.ConntrackZone))
 	if err != nil {
 		return fmt.Errorf("Failed to add openflow flow to %s, stderr: %q, "+
 			"error: %v", gwBridge, stderr, err)
 	}
 
 	// table 1, established and related connections go to pod
 	_, stderr, err = util.RunOVSOfctl("add-flow", gwBridge,
-		fmt.Sprintf("priority=100, table=1, ct_state=+trk+est, "+
-			"actions=output:%s", ofportPatch))
+		fmt.Sprintf("cookie=%s, priority=100, table=1, ct_state=+trk+est, "+
+			"actions=output:%s", util.DefaultOpenFlowCookie, ofportPatch))
 	if err != nil {
 		return fmt.Errorf("Failed to add openflow flow to %s, stderr: %q, "+
 			"error: %v", gwBridge, stderr, err)
 	}
 	_, stderr, err = util.RunOVSOfctl("add-flow", gwBridge,
-		fmt.Sprintf("priority=100, table=1, ct_state=+trk+rel, "+
-			"actions=output:%s", ofportPatch))
+		fmt.Sprintf("cookie=%s, priority=100, table=1, ct_state=+trk+rel, "+
+			"actions=output:%s", util.DefaultOpenFlowCookie, ofportPatch))
 	if err != nil {
 		return fmt.Errorf("Failed to add openflow flow to %s, stderr: %q, "+
 			"error: %v", gwBridge, stderr, err)
 	}
 
 	// table 1, all other connections do normal processing
 	_, stderr, err = util.RunOVSOfctl("add-flow", gwBridge,
-		"priority=0, table=1, actions=output:NORMAL")
+		fmt.Sprintf("cookie=%s, priority=0, table=1, actions=output:NORMAL", util.DefaultOpenFlowCookie))
 	if err != nil {
 		return fmt.Errorf("Failed to add openflow flow to %s, stderr: %q, "+
 			"error: %v", gwBridge, stderr, err)
 	}
+
+	// add health check function to check default OpenFlow flows are on the shared gateway bridge
+	go checkDefaultConntrackRules(gwBridge, stopChan)
```

Comment:
@girishmg can we pass the # of rules to expect into checkDefaultConntrackRules()? I worry that if we need to change that number in the future it'll take 30 minutes to figure out why things are wrong at runtime.